test/models/TaxYearModelSpec.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import models.incomeSourceDetails.TaxYear
+import testUtils.UnitSpec
+
+class TaxYearModelSpec extends UnitSpec {
+
+
+  "TaxYear.getTaxYearStartYearEndYear" when {
```

Comment:
its seems a good use case to use scalaCheck here, as this will shorten your code, but really up to you